scripts/deployments/deploy.sh

```diff
@@ -292,26 +292,40 @@ if [ "$start_simcore" -eq 0 ]; then
     fi
 fi
 # shellcheck disable=2235
-if [ "$start_opsstack" -eq 0 ] && ([ "$stack_target" = "dalco" ] || [ "$stack_target" = "master" ] || [ "$stack_target" = "public" ]); then
+
+# We pause 2 mns to be sure the deployment agent is launched
+sleep 120
+
+
+if ([ "$stack_target" = "dalco" ] || [ "$stack_target" = "master" ] || [ "$stack_target" = "public" ]); then
+
+    # How many deploys do we have ?
+    deploys_count=$(docker service ls --format "{{.Name}}" | grep "deployment-agent" | wc -l)
+    container_count=$(docker ps --format "{{.Names}}" | grep "simcore_.*_postgres" | wc -l)
+
+
     # -------------------------------- BACKUP PG -------------------------------
     # PG-backup has to wait for postgres container to be started and ready before starting, or it will fail.
-    # Wait for potsgres container to start
-    log_info "Before starting PG-backup, we ensure that postgres is started and ready to accept connections."
-    until docker ps -a --format '{{.Names}}' | grep -q "postgres"; do
-      log_info "Postgres didn't start yet. Waiting for 5 seconds..."
-      sleep 5
+    # Wait for all potsgres container to start
+    log_info "Before starting PG-backup, we ensure that all postgres are started and ready to accept connections."
+    until [[ $(docker ps --format "{{.Names}}" | grep "simcore_.*_postgres" | wc -l) -eq $deploys_count ]]; do
```

Comment:
I think in all these calls to `docker ps` you assume that postgres runs on the same node as the one that the scripts runs on. This does not need to be the case. Maybe it would be better to check the status of the docker services relating to postgres?

Reply:
But the script `scripts/deployments/deploy.sh` does not necessarily run on the node that has the label `postgres=true`. So docker ps here wont work

Reply:
Fixed using docker services in place of docker ps.